Fix Promscale installation (#817)
timescale/docs: promscale/installation/kubernetes.md

You can install Promscale on Kubernetes using Helm or using a manifest file.

> ## Install Promscale with Helm

Reviewer: ```suggestion
## Install Promscale with Helm
You can install Promscale using Helm charts.
```

Before you begin, you must have installed Helm. For more information, including
packages and installation instructions, see the

> [Helm documentation][helm-install]

Reviewer: ```suggestion
[Helm documentation][helm-install].
```

> You can install Promscale using Helm charts. The Helm charts must be installed

Reviewer: ```suggestion
The Helm charts must be installed
```

in this order:
1.  Install the TimescaleDB Helm chart
1.  Install the Promscale Helm chart

### Install the TimescaleDB Helm chart
Before you install the TimescaleDB Helm chart, you need to configure these
settings in the [`values.yaml`][timescaledb-single-values-yaml] configuration file:
*   Credentials for the superuser, admin, and other users

> *   TLS Certificates

Reviewer 2: @VineethReddy02 can you take a look at this?

> *   TLS Certificates

Author: I have added a commit, that references these sections in `values.yaml`.

`helm install`.
</highlight>

By default, the `timescaledb-single` helm chart deploys TimescaleDB in 
high availability mode. By default, this creates three database replicas, 
which consumes three times the amount of disk space. Each database 
instance mounts to its own persistent volume claim (PVC). You can 
disable high availability mode by changing the `timescaledb-single` 

> helm chart `replicaCount` values to `1` in ` values.yaml`.

Reviewer 2: @VineethReddy02 where will users find the values.yaml file?

`helm install`.
</highlight>

> By default, `timescaledb-single` helm chart deploys the TimescaleDB in **high-availability** mode. This results in running 3 replicas of database instances which in return comsumes 3x of disk as each database instance mounts to it's own persistent volume claim (PVC). You can disable high-availability mode by changing the `timescaledb-single` helm chart `replicaCount` values to `1` in ` values.yaml`.

Reviewer 3: ```suggestion
By default, the `timescaledb-single` helm chart deploys TimescaleDB in 
high availability mode. By default, this creates three database replicas, 
which consumes three times the amount of disk space. Each database 
instance mounts to its own persistent volume claim (PVC). You can 
disable high availability mode by changing the `timescaledb-single` 
helm chart `replicaCount` values to `1` in ` values.yaml`.
```

Reviewer 2: @VineethReddy02 where will a user find the the values.yaml file? Is there some extra information we could provide to help them locate the file?

`helm install`.
</highlight>

> By default, the `timescaledb-single` helm chart deploys TimescaleDB in 

Reviewer: ```suggestion
By default, the `timescaledb-single` Helm chart deploys TimescaleDB in 
```

> high availability mode. By default, this creates three database replicas, 

Reviewer: ```suggestion
high availability mode. This creates three database replicas, 
```

which consumes three times the amount of disk space. Each database 
instance mounts to its own persistent volume claim (PVC). 

You can 

> disable high availability mode by changing the `timescaledb-single` 
> helm chart `replicaCount` values to `1` in [`values.yaml`][timescaledb-single-values-yaml].

Reviewer: ```suggestion
disable high availability mode by changing the value of `replicaCount`
to `1` in [`values.yaml`][timescaledb-single-values-yaml].
```

<procedure>

> #### Steps to disbale TimescaleDB High-Availability

Reviewer: ```suggestion
#### Disabling TimescaleDB high availability mode
```

> 1.  Download the default `values.yaml` of `timescaledb-single` helm chart from [here][timescaledb-single-values-yaml].

Reviewer: ```suggestion
1.  Download the default `values.yaml` for the `timescaledb-single` Helm chart. To get the
file, see the [`timescaledb-kubernetes` repository][timescaledb-single-values-yaml].
```

Reviewer 4: I updated this some @charislam. We aren't actually directing them to the repository so the second sentence didn't make sense and this is just direct. "download the file..."

> 1.  In `values.yaml` change the default `replicaCount` from `3` to `1`. 

Reviewer: ```suggestion
1.  In `values.yaml`, change the default `replicaCount` from `3` to `1`. 
```

> 1.  In the next step while installing the `timescaledb-single` helm chart, use this `values.yaml` file with `-f` flag.

Reviewer: ```suggestion
1.  Use this `values.yaml` file with the `-f` flag when installing the `timescaledb-single` Helm chart.
    For installation instructions, see [the procedures for installing the Helm chart](#installing-the-timescaledb-helm-chart).
```

    ```bash
    helm repo update
    ```

> 1.  Create a database with name `tsdb` for Promscale data

Reviewer 3: ```suggestion
1.  Create a database called `tsdb` for Promscale data:
```

    ```bash
    MASTERPOD="$(kubectl get pod -o name --namespace <namespace> -l release=<releaseName>,role=master)"
    kubectl exec -i --tty --namespace default ${MASTERPOD} -- psql -U postgres
    CREATE DATABASE tsdb WITH OWNER postgres;
    ```

> 1.  Capture the `postgres` user password using

Reviewer 3: ```suggestion
1.  Capture the `postgres` user password:
```

    ```bash
    helm repo update
    ```
1.  Create a database called `tsdb` for Promscale data:
    ```bash

>     MASTERPOD="$(kubectl get pod -o name --namespace <namespace> -l release=<releaseName>,role=master)"

Reviewer: ```suggestion
    MASTERPOD="$(kubectl get pod -o name --namespace <NAMESPACE> -l release=<RELEASE_NAME>,role=master)"
```

    kubectl exec -i --tty --namespace default ${MASTERPOD} -- psql -U postgres

>     CREATE DATABASE tsdb WITH OWNER postgres;

Reviewer: Should these be made into separate steps?

Reviewer 4: I don't think so @VineethReddy02, correct? However, as shown, I believe we'll need the `\` at the end of each line.

Essentially, the "MASTERPOD" is a variable passed into the next command, and the `psql` command what we're asking the `kubectl exec` to run.

But again, if someone copied and pasted these three separate lines, it wouldn't work without the `\` on the first two lines to signal continuation.

Reviewer 2: That's my understanding as well. @VineethReddy02 please confirm.

Author: I have slightly updated these instructions and I have tested them from my end, works fine. These instructions are taken from here in upstream [timescaledb helm docs](https://github.com/timescale/timescaledb-kubernetes/tree/master/charts/timescaledb-single#connecting-from-inside-the-cluster) and they do not need `\` explicitly at the end of each instruction.

    ```
1.  Capture the `postgres` user password:
    ```bash
    echo $(kubectl get secret --namespace default tobs-credentials -o jsonpath="{.data.PATRONI_SUPERUSER_PASSWORD}" | base64 --decode)
    ```
1.  Download the Promscale

>     `values.yaml` from [here][promscale-values-yaml], open the `values.yaml` configuration file, and locate the `connection`

Reviewer 4: @charislam @Rajakavitha1 - whatever we choose to do for linking/wording on the TimescaleDB procedure above, we need to do the same thing here. I personally don't mind what @VineethReddy02 did here (linking to the word 'here'), but it's not consistent with the changes above. Decide on one of the two.

Reviewer 5: Hi @VineethReddy02  and @ryanbooz  Here's what the style guide recommends: 
"Write unique, descriptive link text that makes sense without the surrounding text. Don't use phrases such as this document, this article, or click here."

Reviewer 2: What about something like

1.  [Download the Promscale `values.yaml` file][promscale-values-yaml] and edit  the `connection` section in that file with your TimescaleDB connection details:

Reviewer 5: > What about something like
> 
> 1. [Download the Promscale `values.yaml` file][promscale-values-yaml] and edit  the `connection` section in that file with your TimescaleDB connection details:

Reviewer 5: I made a suggestion. Please feel free to make or ignore the change we can always work on editing after the content is published.

Reviewer: Applied your suggestion @Rajakavitha1, I think it's consistent now

    </tab>

    OR

    <tab label="Connection parameters">

    ```yaml
    connection:
      user: postgres
      password: ""

>       host: <releaseName>.<namespace>.svc.cluster.local

Reviewer 3: ```suggestion
      host: <RELEASE_NAMEW.<NAMESPACE>.svc.cluster.local
```
